pkg/cli/cluster_free_disk_space.go

```diff
@@ -0,0 +1,133 @@
+package cli
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+
+	"github.com/spf13/cobra"
+	v1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	rookcli "github.com/rook/rook/pkg/client/clientset/versioned"
+
+	clusterspace "github.com/replicatedhq/kurl/pkg/cluster/space"
+)
+
+const (
+	// defaultOpenEBSPodImage is the image used during disk free check for openebs storage.
+	// this is the same image used by the pvmigrate project, it may be any image containing
+	// 'df' and 'cat' commands.
+	defaultOpenEBSPodImage          = "eeacms/rsync:2.3"
+	isDefaultStorageClassAnnotation = "storageclass.kubernetes.io/is-default-class"
+	openEBSLocalProvisioner         = "openebs.io/local"
+	rookRBDProvisioner              = "rook-ceph.rbd.csi.ceph.com"
+	rookCephFSProvisioner           = "rook-ceph.cephfs.csi.ceph.com"
+)
+
+// NewClusterCheckFreeDiskSpaceCmd returns a command that is capable of reporting back the amount of free space in the cluster
+// for a provided storage class.
+func NewClusterCheckFreeDiskSpaceCmd(cli CLI) *cobra.Command {
+	var forStorageClass, openEBSImage, openEBSNode string
+	logbuf := bytes.NewBuffer(nil)
+	logger := log.New(logbuf, "", 0)
+	k8sConfig := config.GetConfigOrDie()
+	clientSet := kubernetes.NewForConfigOrDie(k8sConfig)
+	rookClientSet := rookcli.NewForConfigOrDie(k8sConfig)
```

Comment:
can you instantiate these in the run() function?

Reply:
IHMO it should to be PreRunE since is a pre-requirement as suggested https://github.com/replicatedhq/kURL/pull/3784#discussion_r1034843469

I know that the methods here will exit(1) if fails but I also think that should be inside of the command so that if errors occurs we cobra would get that. ( i.e it is within silenceUsage is true but if want to remove that, cobra would output the usage/help for the command as well )

Reply:
im fine with prerune. i just dont like that to get the help text you have to be able to instantiate the kube client